### PR TITLE
upgrade: vala from 0.54.6 to 0.56.14

### PR DESCRIPTION
**Pull request by SeanDougherty**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

###### Change Log  

- Upgrade from 0.54.6 to 0.56.14

###### Does this affect the toolchain?  

NO

###### Test Methodology

- local build
- Current buddy build - [Link](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=501152&view=results) - Built successfully with graphviz fix from #7728

### CODE REVIEW

File: cgmanifest.json [hunk 1]
```diff
@@ -28967,8 +28967,8 @@
         "type": "other",
         "other": {
           "name": "vala",
-          "version": "0.54.6",
-          "downloadUrl": "http://download.gnome.org/sources/vala/0.54/vala-0.54.6.tar.xz"
+          "version": "0.54.14",
```
Comment (anphel31) [inline cgmanifest.json:28970]: should be `0.56.14` and `https://download.gnome.org/sources/vala/0.56/vala-0.56.14.tar.xz`

**Review by anphel31 — COMMENTED** [review]
Can you please run a buddybuild and link in the description

**Comment by SeanDougherty** [comment]
> Can you please run a buddybuild and link in the description

https://dev.azure.com/mariner-org/mariner/_build/results?buildId=500543&view=results

**Comment by nisamson** [comment]
Build failed due to issues with a dependency; gonna hold off on approval for now.

**Comment by SeanDougherty** [comment]
> Can you please run a buddybuild and link in the description

Thank you @anphel31 , your feedback has been incorporated! Could I get your follow-up review?

**Review by arc9693 — APPROVED** [review]
LGTM

**Review by BettyRain — APPROVED** [review]

**Pull request merged**